src/main/java/codeu/controller/ProfilePageServlet.java

```diff
@@ -53,10 +58,33 @@ void setUserProfileStore(UserProfileStore userProfileStore) {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) 
     throws IOException, ServletException { 
-    /* TODO 
-     * Retrieve UserProfile data
-     * Fill ProfilePage fields with retrieved data
-     */
+
+    String username = (String)request.getSession().getAttribute("user"); 
+    if(username == null){
+      response.sendRedirect("/profilepage");
```

Comment:
Maybe if the user doesn't have a username (which means they haven't logged in) we should redirect them to the login page?

Reply:
I was thinking we really don't even need that check because the link to the profile won't even be active or viewable if they aren't logged in under a valid user...

Reply:
The `request.setAttribute("error", "error message...")` used elsewhere in the code could also be an option here.

Reply:
Just tested it out, I never even thought of the possibility of someone using the URL when they aren't logged in *face palm* If any one's curious, if you redirect it to the profile page it just says it can't find the page. I changed it to the login page and it will redirect you to that successfully now!